contracts/integrations/test/framework-unit-tests/getter_cache_test.ts

```diff
@@ -0,0 +1,158 @@
+import { blockchainTests, constants, expect, hexSlice } from '@0x/contracts-test-utils';
+import { BigNumber } from '@0x/utils';
+import { BlockParam, CallData } from 'ethereum-types';
+import * as ethUtil from 'ethereumjs-util';
+
+import { artifacts, TestFrameworkContract } from '../../src';
+import { GetterCache } from '../utils/cache';
+
+blockchainTests.resets('Cache Tests', env => {
+    let exampleContract: TestFrameworkContract;
+
+    before(async () => {
+        exampleContract = await TestFrameworkContract.deployFrom0xArtifactAsync(
+            artifacts.TestFramework,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    describe('callAsync', () => {
+        describe('() => uint', () => {
```

Comment:
I wanted to write all of these tests for my benefit just to ensure that all of the types actually work with the the cache. I'd prefer to keep these tests so that the behavior of the cache doesn't regress across any of these types